obspy/io/scardec/core.py

```diff
@@ -0,0 +1,457 @@
+        warnings.warn("No moment wave magnitude found. Will be replaced by the"
+                      " first magnitude in the event object.")
+        mw_mag = event.magnitudes[0]
+    else:
+        mw_mag = mw_candidates[0]
+
+    # Now find the cmt origin. Try to find the first one that is CMT
+    candidates = [_i for _i in event.origins
+                  if _i.origin_type == "centroid"]
+    if candidates:
+        warnings.warn("No derived origin attached to the moment tensor. "
+                      "Will instead use another centroid origin to be "
+                      "written to the file.")
+        cmt_origin = candidates[0]
+    # Otherwise just take the preferred or first one.
+    else:
+        warnings.warn("Could not find a centroid origin. Will instead "
+                      "assume that the preferred or first origin is the "
+                      "centroid origin.")
+        cmt_origin = event.preferred_origin() or event.origins[0]
+
+    if not foc_mec.moment_tensor.source_time_function.extra:
+        raise ValueError('Event focal mechanism must contain a source time \
+                          function to be written in SCARDEC format')
+
+    # Now attempt to retrieve the event name. Otherwise just get a random one.
+    event_name = None
+    if event.event_descriptions:
+        candidates = [_i for _i in event.event_descriptions
+                      if _i.type == "earthquake name"]
+        if candidates:
+            event_name = candidates[0].text
+    if event_name is None:
+        event_name = str(uuid.uuid4())[:6].upper()
+
+    template = (
+        "{year:4d} {month:02d} {day:02d} {hour:02d} "
+        "{minute:02d} {second:04.1f} "
+        "{latitude:9.4f} {longitude:9.4f}\n"
+        "{depth:5.1f} {scalmom:9.3E} {mw:5.3f}"
+        "{strike1:4d} {dip1:4d} {rake1:4d}"
+        "{strike2:4d} {dip2:4d} {rake2:4d}\n"
+    )
+
+    np1 = foc_mec.nodal_planes.nodal_plane_1
+    np2 = foc_mec.nodal_planes.nodal_plane_2
+
+    template = template.format(
+        year=cmt_origin.time.year,
+        month=cmt_origin.time.month,
+        day=cmt_origin.time.day,
+        hour=cmt_origin.time.hour,
+        minute=cmt_origin.time.minute,
+        second=float(cmt_origin.time.second) +
+        cmt_origin.time.microsecond / 1E6,
+        latitude=cmt_origin.latitude,
+        longitude=cmt_origin.longitude,
+        depth=cmt_origin.depth / 1000.0,
+        scalmom=foc_mec.moment_tensor.scalar_moment,
+        mw=mw_mag.mag,
+        strike1=int(np1.strike),
+        dip1=int(np1.dip),
+        rake1=int(np1.rake),
+        strike2=int(np2.strike),
+        dip2=int(np2.dip),
+        rake2=int(np2.rake)
+    )
+
+    # Write to a buffer/file opened in binary mode.
+    buf.write(template.encode())
+
+    stf = foc_mec.moment_tensor.source_time_function.extra
+    t = stf['offset']['value']
+    scalmom = foc_mec.moment_tensor.scalar_moment
+    for sample in stf['moment_rate']['value']:
```

Comment:
This could in principle probably be done much faster using `np.savetxt(buf, ...)`, avoiding the loop. But it would only matter when writing lots of files I guess..